python/semantic_kernel/contents/binary_content.py

```diff
@@ -105,27 +114,33 @@ def data_uri(self) -> str:
     @data_uri.setter
     def data_uri(self, value: str):
         """Set the data uri."""
-        self._data_uri = DataUri.from_data_uri(value)
+        if not self._data_uri:
+            self._data_uri = DataUri.from_data_uri(value, self.default_mime_type)
+        else:
+            self._data_uri.update_data(value)
         self.metadata.update(self._data_uri.parameters)
 
     @property
     def data(self) -> bytes:
         """Get the data."""
+        if self._data_uri and self._data_uri.data_array:
+            return self._data_uri.data_array.tobytes()
         if self._data_uri and self._data_uri.data_bytes:
             return self._data_uri.data_bytes
-        if self._data_uri and self._data_uri.data_str:
-            return self._data_uri.data_str.encode("utf-8")
         return b""
 
     @data.setter
-    def data(self, value: str | bytes):
+    def data(self, value: str | bytes | ndarray):
         """Set the data."""
         if self._data_uri:
             self._data_uri.update_data(value)
-        else:
-            if isinstance(value, str):
+            return
+        match value:
+            case ndarray():
+                self._data_uri = DataUri(data_array=value, mime_type=self.mime_type)
+            case str():
                 self._data_uri = DataUri(data_str=value, mime_type=self.mime_type)
-            else:
+            case bytes():
```

Comment:
No else handling needed here? No exception needed if we get a type that isn't `str | bytes | ndarray`?

Reply:
good point!